PLEP-0007.rst

```diff
@@ -0,0 +1,847 @@
+
+.. [Poli2018] F. M. Poli (2018), `*Integrated Tokamak modeling: When
+   physics informs engineering and research planning*
+   <https://doi.org/10.1063/1.5021489>`_, Physics of Plasmas, 25,
+   055602, doi: 10.1063/1.5021489
+
+Modeling the solar chromosphere
+-------------------------------
+
+The solar chromosphere hosts a rich variety of physical processes.
+Many of the simplifying assumptions that are valid in either the
+photosphere below or the corona above cannot be applied to the
+chromosphere [e.g., the assumption of local thermodynamic equilibrium
+(LTE)].  The plasma is partially ionized, so interactions between
+neutrals and charged particles are important.  Non-LTE radiative
+transfer is required.  Shocks contribute to the heating.  Some solar
+physicists will want to model a beam of energetic electrons
+precipitating from the corona into the chromosphere during solar
+flares. Synthetic observables will be required to validate simulation
+results against reality.
+
+Software Requirements Specification
+===================================
+
+Availability requirements
+-------------------------
+
+* Develop openly under the Open Source Initiative approved `BSD+Patent
+  <https://opensource.org/licenses/BSDplusPatent>`_ license.
+
+* Allow installation in a single line with widely-used package
+  managers.
+
+  - Do not require users to compile external libraries, edit make
+    files, change compiler flags, or run bash scripts.
+
+  - Depend only on packages that can be installed using that package
+    manager.
+
+* Design the package to run in Linux, macOS, and Windows environments
+  and on scales ranging from a single core to the most powerful
+  supercomputers.
+
+Language requirements
+---------------------
+
+* Develop the base functionality using Julia.
+
+  - Do not include any statically typed or non-interactive languages
+    because that will decrease long-term maintainability and because
+    mixing languages can cause problems with compilers.
+
+* Provide a Python interface to the core functionality.
+
+  - Include this interface in PlasmaPy or an affiliated package in the
+    PlasmaPy ecosystem.
+
+Application programming interface (API) requirements
+----------------------------------------------
+
+* Create APIs that enable multiple levels of abstraction.  Greater
+  abstraction will make the code easier to use, whereas less
+  abstraction will provide more options for customizability and
+  fine-tuning.
+
+  - Enable a simple simulation to be set up in ten to twenty lines of
+    code.
+
+  - Provide narrative documentation that progresses from a simulation
+    set up using the highest level of abstraction to lower levels of
+    abstraction.
+
+* Each numerical method should be programmed to a well-defined
+  interface.  PIC and fluid simulations should have identical
+  interfaces at the highest level of abstraction.
+  
+Functionality requirements
+--------------------------
```

Comment:
TODO Note to self: pick up from here! Sorry Nick, but this got longer than my break allowed me to read!